### PR TITLE
Refactors to draft/live functionality and associated tests

### PR DESCRIPTION
**Pull request by lfdebrux**
While working on https://trello.com/c/845sNCtW I found there were a bunch of things that it would be good to clean up in our models to make it easy to tell when a form had a draft version and/or a live version.

I've split those changes out into a separate PR, to break things up a bit.

#### What problem does the pull request solve?

- It was hard to write model specs that change a form or a page while doing the same bookkeeping as the controllers, we move that business logic into a new `FormUpdateService`
- The datetime stamps for live_at and made_live_forms.last weren't consistent with the updated_at of the form, so we couldn't compare reliably
- The logic for creating a JSON form blob was scattered all over in our code and in our specs, we restore the `snapshot` method to bring it all into one place

Those are the major ones, there are also some small tweaks I snuck in as well ;)

### CODE REVIEW

File: app/controllers/api/v1/pages_controller.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def show
   def update
     page.assign_attributes(page_params)
 
-    if page.save
+    if FormUpdateService.save_page(page)
```
Comment (aliuk2012) [inline app/controllers/api/v1/pages_controller.rb]: Not sure I follow the difference here,

File: app/models/form.rb [hunk 2]
```diff
@@ -20,6 +20,10 @@ def draft_version
     snapshot.to_json
   end
 
+  def live_at
+    return made_live_forms.last.created_at
+  end
```
Comment (aliuk2012) [inline app/models/form.rb]: Could we add a test for this?

File: app/service/form_update_service.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      form.save
+    end
+
+    def update(form, **options)
+      form.update(**options)
+    end
+
+    def destroy!(form)
+      form.destroy!
+    end
```
Comment (aliuk2012) [inline app/service/form_update_service.rb:13]: not sure I'm following this service. It feels like we just trying to duplicate what ActiveRecord already gives us.

Comment (lfdebrux) [reply app/service/form_update_service.rb]: @thomasiles @aliuk2012 and I discussed and agreed a different approach, see [64290fa](https://github.com/alphagov/forms-api/pull/232/commits/64290fa4f14a8225abe7319025a7eaddc68682a5).

File: app/models/form.rb (hunk 2 above)
Comment (lfdebrux) [reply app/models/form.rb]: Done!

File: app/controllers/api/v1/pages_controller.rb (hunk 1 above)
Comment (lfdebrux) [reply app/controllers/api/v1/pages_controller.rb]: @thomasiles @aliuk2012 and I discussed and agreed a different approach, see [f1061ed](https://github.com/alphagov/forms-api/pull/232/commits/f1061edc2712004b5f1532e8982fae9bf901c3bd).

**Review by aliuk2012 — APPROVED** [review]
Some good refactors and I like the reintroduction of "snapshots". 👍

**Pull request merged**